src/RevProc.cc

```diff
@@ -2372,8 +2372,10 @@ void RevProc::ExecEcall(RevInst& inst){
 
     // For now, rewind the PC and keep executing the ECALL until we
     // have completed
-    if(EcallStatus::SUCCESS != status){
+    if( status != EcallStatus::SUCCESS ){
       RegFile->SetPC( RegFile->GetPC() - inst.instSize );
+    } else {
+      Harts[HartToDecodeID]->GetEcallState().clear();
```

Comment:
This is probably a good general solution for all `ECALL`s, but we already call `EcallState.clear()` on success in `rev_write`:

```c++
  if(nleft == 0 && LSQueue->count(lsq_hash) == 0){
    int rc = write(fd, EcallState.string.data(), EcallState.string.size());
    RegFile->SetX(RevReg::a0, rc);
    EcallState.clear();
    DependencyClear(HartToExecID, RevReg::a0, false);
    return EcallStatus::SUCCESS;
}
```